lib/phoenix_component/declarative.ex

```diff
@@ -512,16 +512,19 @@ defmodule Phoenix.Component.Declarative do
     """)
   end
 
-  defp valid_value?(_type, nil), do: true
-  defp valid_value?(:any, _value), do: true
-  defp valid_value?(:string, value), do: is_binary(value)
-  defp valid_value?(:atom, value), do: is_atom(value)
-  defp valid_value?(:boolean, value), do: is_boolean(value)
-  defp valid_value?(:integer, value), do: is_integer(value)
-  defp valid_value?(:float, value), do: is_float(value)
-  defp valid_value?(:list, value), do: is_list(value)
-  defp valid_value?({:struct, mod}, value), do: is_struct(value, mod)
-  defp valid_value?(_type, _value), do: true
+  def valid_value?(_type, nil), do: true
+  def valid_value?({:struct, _mod}, nil), do: true
```

Comment:
I must be missing something, but given these two lines in this order:
```
def valid_value?(_type, nil), do: true
def valid_value?({:struct, _mod}, nil), do: true
```
how does the 2nd line ever get matched?  As long as the 2nd argument is `nil` the first function implementation will match.  Right?  I just don't see how this change alters anything except the public visibility of `valid_value?/2` and matching some more types further down (eg `:map`, `:fun` and `{:fun, arity}`.  Apologies if i'm just being ignorant here.

Reply:
You’re right, this doesn’t change anything. The inconsistency in #3790 comes from LiveView generating a special function head for structs.